File: Python/102_Binary_Tree_Level_Order_Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res={}
        def helper(node,depth):
            if node is None:
                return
            if depth not in res:
                res[depth]=[node.val]
            else:
                res[depth].append(node.val)
            if node.left: 
                helper(node.left,depth+1)
            if node.right: 
                helper(node.right,depth+1)
        helper(root,0)
        return [res[depth] for depth in res]

File: Java/102_Binary_Tree_Level_Order_Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q=new LinkedList<>();   
        List<List<Integer>> res= new ArrayList<>();
        q.add(root);
        while(!q.isEmpty()){
            List<Integer> l = new ArrayList<>();
            int size=q.size();
            for(int i=0;i<size;i++){
                TreeNode node=q.poll();
                if (node!=null){
                    l.add(node.val);
                    q.add(node.left);
                    q.add(node.right);                    
                }
            }
            if(l.size()>0){
                res.add(l);
            }
        }
        return res;
    }
}
